larq_compute_engine/core/bgemm_kernels_common.h

```diff
@@ -1,61 +1,23 @@
 #ifndef COMPUTE_EGNINE_TFLITE_KERNELS_BGEMM_KERNELS_COMMON_H_
 #define COMPUTE_EGNINE_TFLITE_KERNELS_BGEMM_KERNELS_COMMON_H_
 
-#include <cstdint>
-
+#include "larq_compute_engine/core/bconv2d_output_transform.h"
 #include "tensorflow/lite/experimental/ruy/kernel_common.h"
-#include "tensorflow/lite/kernels/cpu_backend_gemm_params.h"
 
 using namespace ruy;
 
-using tflite::cpu_backend_gemm::QuantizationFlavor;
-
-// Our version of `cpu_backend_gemm::GemmParams`
-// Original is in `lite/kernels/cpu_backend_gemm_params.h`
-// Modifications:
-// - bias changes to multiply + add
-// - clamp_min, clamp_max type changed from DstScalar to AccumScalar
-// - (later) 8-bit quantization related stuff
-template <typename AccumScalar, typename DstScalar,
-          QuantizationFlavor quantization_flavor =
-              std::is_floating_point<DstScalar>::value
-                  ? QuantizationFlavor::kFloatingPoint
-                  : QuantizationFlavor::kIntegerWithUniformMultiplier>
-struct BGemmParams {
-  AccumScalar multiplier_fixedpoint = 0;
-  int multiplier_exponent = 0;
-  std::int32_t backtransform_add = 0;
-  // post_mutiply and post_activation_bias are currently float
-  // in order to accomodate for batchnorm scales
-  // Later this might be changed to the int8 system of multipliers+shifts
-  const float* post_activation_multiplier = nullptr;
-  const float* post_activation_bias = nullptr;
-  AccumScalar clamp_min = std::numeric_limits<AccumScalar>::lowest();
-  AccumScalar clamp_max = std::numeric_limits<AccumScalar>::max();
-};
+using compute_engine::core::OutputTransform;
 
-//
 // Our version of `ruy::BasicSpec`
 // Original is in `lite/experimental/ruy/spec.h`
-// Modifications:
-// - bias changes to multiply + add
-// - clamp_min, clamp_max types changed from DstScalar to AccumScalar
-// - (later) 8-bit quantization related stuff
-
+// We simply subclass our `OutputTransform` struct
 template <typename tAccumScalar, typename tDstScalar>
-struct BinaryBasicSpec {
+struct BinaryBasicSpec : OutputTransform<tAccumScalar, tDstScalar> {
```

Comment:
if we have OutputTransform as a member of BinaryBasicSpec instead of subclassing, we can simplify the code quite more.